### PR TITLE
fixed issue and refactored admin bar code

### PR DESCRIPTION
**Pull request by lucatume**
Ticket https://central.tri.be/issues/41021
I've fixed the issue and refactored (with tests) admin bar related code out of Main.

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | There must be a single space before the last parenthesis | [Admin_Bar.php:52](https://github.com/moderntribe/the-events-calendar/blob/e8b4cb7a5a33fccf906abad31e036ed80748c6b1/src/Tribe/Admin/Bar/Admin_Bar.php#L52)
:x: | ! must be followed by a single space | [Admin_Bar.php:52](https://github.com/moderntribe/the-events-calendar/blob/e8b4cb7a5a33fccf906abad31e036ed80748c6b1/src/Tribe/Admin/Bar/Admin_Bar.php#L52)
:x: | There must be a single space before the last parenthesis | [Admin_Bar.php:52](https://github.com/moderntribe/the-events-calendar/blob/e8b4cb7a5a33fccf906abad31e036ed80748c6b1/src/Tribe/Admin/Bar/Admin_Bar.php#L52)
:x: | There must be a single space after the first parenthesis | [Configurator_Interface.php:13](https://github.com/moderntribe/the-events-calendar/blob/e8b4cb7a5a33fccf906abad31e036ed80748c6b1/src/Tribe/Admin/Bar/Configurator_Interface.php#L13)
:x: | There must be a single space after the first parenthesis | [Configurator_Interface.php:13](https://github.com/moderntribe/the-events-calendar/blob/e8b4cb7a5a33fccf906abad31e036ed80748c6b1/src/Tribe/Admin/Bar/Configurator_Interface.php#L13)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Constants.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	protected $volatile_values;
+
+	/**
+	 * Tribe__Events__Constants constructor.
+	 *
+	 * @param bool $volatile If `true` the class will not define and read real constants.
+	 */
+	public function __construct( $volatile = false ) {
+		$this->volatile        = $volatile;
+		$this->volatile_values = [ ];
```
Comment (borkweb) [inline src/Tribe/Constants.php]: To support earlier versions of PHP, switch to `array();`

File: tests/functional/Tribe/Events/Admin/Bar/Admin_BarTest.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+		$sut = new Admin_Bar( $configurator->reveal(), $constants );
+		$sut->init( $admin_bar );
+
+		$configurator->configure( $admin_bar )->shouldHaveBeenCalled();
+	}
+}
+
+
+class Dummy_WP_Screen {
```
Comment (borkweb) [inline tests/functional/Tribe/Events/Admin/Bar/Admin_BarTest.php]: Move this to its own file?

Comment (borkweb) [inline tests/functional/Tribe/Events/Admin/Bar/Admin_BarTest.php]: I mean, I can see why that feels silly. But nestling it next to another class is also kind of wonky.

File: src/Tribe/Main.php [hunk 3]
```diff
@@ -4089,136 +4089,13 @@ public static function postIdHelper( $postId = null ) {
...
-						);
-					}
-				}
+		public function add_toolbar_items() {
+			$admin_bar = Tribe__Events__Admin__Bar__Admin_Bar::instance();
+			if ( ! $admin_bar->is_enabled() ) {
+				return;
 			}
+			global $wp_admin_bar;
+			$admin_bar->init( $wp_admin_bar );
```
Comment (borkweb) [inline src/Tribe/Main.php]: Man, this change makes me so happy :)

File: tests/functional/Tribe/Events/Admin/Bar/Admin_BarTest.php (hunk 2 above)
Comment (lucatume) [inline tests/functional/Tribe/Events/Admin/Bar/Admin_BarTest.php]: This is a good candidate for a Tribe Test utility repo, will come in handy elsewhere and there is no reason to duplicate the code

**Comment by borkweb** [comment]
:+1:

**Pull request merged**